container/start.sh

```diff
@@ -1,149 +1,80 @@
-broker.id=0
-listener.security.protocol.map=PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT,PLAINTEXT_ALT:PLAINTEXT
-advertised.listeners=${adv_listeners}
-listeners=${listeners}
-num.network.threads=3
-num.io.threads=8
-socket.send.buffer.bytes=102400
-socket.receive.buffer.bytes=102400
-socket.request.max.bytes=104857600
-log.dirs=/var/lib/kafka
-num.partitions=1
-num.recovery.threads.per.data.dir=1
-offsets.topic.replication.factor=1
-transaction.state.log.replication.factor=1
-transaction.state.log.min.isr=1
-log.retention.hours=168
-log.segment.bytes=1073741824
-log.retention.check.interval.ms=300000
-zookeeper.connect=${zk_host}:${zk_port}
-zookeeper.connection.timeout.ms=6000
-group.initial.rebalance.delay.ms=0
-EOF
-  mkdir -p /var/lib/kafka
-  /opt/wait-for-it.sh "${zk_host}:${zk_port}"
-  /opt/kafka_2.12-2.4.1/bin/kafka-server-start.sh /opt/server.properties 2>&1 | tee /var/log/kafka.log
-}
+KARAPACE_REST_PORT_DEFAULT=8082
+KARAPACE_REST_HOST_DEFAULT=0.0.0.0
+# Variables without defaults:
+# KARAPACE_REST_ADVERTISED_HOSTNAME
+# KARAPACE_REST_BOOTSTRAP_URI
+# KARAPACE_REST_REGISTRY_HOST
+# KARAPACE_REST_REGISTRY_PORT
 
 start_karapace_registry(){
   echo "starting karapace schema registry"
 
-  if [ "$1" == "single" ]; then
-    kafka_host="kafka:${kafka_port_internal}"
-    advertised_host="${ADVERTISED_HOST:-registry}"
-  else
-    kafka_host="127.0.0.1:${kafka_port}"
-    advertised_host="127.0.0.1"
-  fi
-  cat >/opt/karapace.config.json <<- EOF
+  cat >/opt/karapace/registry.config.json <<- EOF
 {
-    "advertised_hostname": "${advertised_host}",
-    "bootstrap_uri": "${kafka_host}",
-    "client_id": "sr-1",
+    "advertised_hostname": "${KARAPACE_REGISTRY_ADVERTISED_HOSTNAME}",
+    "bootstrap_uri": "${KARAPACE_REGISTRY_BOOTSTRAP_URI}",
+    "host": "${KARAPACE_REGISTRY_HOST:-$KARAPACE_REGISTRY_HOST_DEFAULT}",
+    "port": ${KARAPACE_REGISTRY_PORT:-$KARAPACE_REGISTRY_PORT_DEFAULT},
+    "client_id": "${KARAPACE_REGISTRY_CLIENT_ID:-$KARAPACE_REGISTRY_CLIENT_ID_DEFAULT}",
+    "group_id": "${KARAPACE_REGISTRY_GROUP_ID:-$KARAPACE_REGISTRY_GROUP_ID_DEFAULT}",
+    "master_eligibility": ${KARAPACE_REGISTRY_MASTER_ELIGIBITY:-$KARAPACE_REGISTRY_MASTER_ELIGIBITY_DEFAULT},
+    "topic_name": "${KARAPACE_REGISTRY_TOPIC_NAME:-$KARAPACE_REGISTRY_TOPIC_NAME_DEFAULT}",
     "compatibility": "FULL",
-    "group_id": "schema-registry",
-    "host": "0.0.0.0",
     "log_level": "INFO",
-    "port": ${registry_port},
-    "master_eligibility": true,
     "replication_factor": 1,
     "security_protocol": "PLAINTEXT",
     "ssl_cafile": null,
     "ssl_certfile": null,
-    "ssl_keyfile": null,
-    "topic_name": "_schemas"
+    "ssl_keyfile": null
 }
 EOF
-  /opt/wait-for-it.sh "${kafka_host}"
-  python3 -m karapace.schema_registry_apis /opt/karapace.config.json 2>&1 | tee /var/log/karapace-registry.log
+  exec python3 -m karapace.schema_registry_apis /opt/karapace/registry.config.json 2>&1 | tee /var/log/karapace/registry.log
```

Comment:
Do we really need to output logs to a file? We also don't have any log rotation, etc.

Reply:
Also if removing this, I'm not 100% sure about `2>&1` - Docker should merge stdout and stderr itself.